public/sw.js

```diff
@@ -0,0 +1,53 @@
+var urlsToCache = [
+  '/',
+  'assets/css/main.css',
+  'assets/img/pattern_130.gif',
+  'assets/js/main.js',
+  'assets/vendor/bootstrap/bootstrap.js',
+  'assets/vendor/bootstrap/bootstrap.min.css',
+  'assets/vendor/prism/prism.js',
+  'assets/vendor/prism/prism.css',
+  'assets/vendor/bootstrap.modalform.js',
+  'assets/vendor/jquery-2.1.0.min.js',
+  'hoodie/client.js',
+  '2.html',
+  '3.html',
+  '4.html',
+  '5.html',
+  '6.html',
+  '7.html'
+];
+
+// 'install' is fired when the ServiceWorker is installed.
+// This only happens once, when the web browser sees this version
+// of the ServiceWorker for the first time.
+self.addEventListener('install', function(event) {
+  // Open a cache to store assets
+  event.waitUntil(
+    caches.open('hoodie-camp-v1').then(function(cache) {
+      return cache.addAll(urlsToCache);
+    })
+  );
+});
+
+
+// The fetch event happens for the page request with the
+// ServiceWorker's scope, and any request made within that
+// page.
+self.addEventListener('fetch', function(event) {
+  event.respondWith(
+    // Is there something in the cache that matches the request?
+    // Return it!
+    // Otherwise pass the request to `fetch`,
+    // which will use the network.
+    caches.match(event.request).then(function(response) {
+      return response || fetch(event.request);
+    })
+  );
+});
+
+// 'activate' is fired when the service worker is ready to handle
+// events.
+self.addEventListener('activate', function(event) {
+  console.log('active! now you can work offline');
```

Comment:
@nickcolley yes! we wanted to tell the users they were offline, but we haven't yet decided how to do it, so I just left this `console.log` to show where to do it. I can remove it before merging.